src/test/java/com/iabtcf/decoder/ConsentParserV1Test.java

```diff
@@ -0,0 +1,185 @@
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import java.time.Instant;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.junit.Test;
+
+public class ConsentParserV1Test {
+
+    private static ConsentParser parse(String consentString) {
+        ConsentParser model = ConsentParser.parse(consentString);
+        assertTrue(model instanceof ConsentParserV1);
+        assertEquals(1, model.getVersion());
+
+        return model;
+    }
+
+    @Test
+    public void testDecodeCanDetectVersion1() {
+        ConsentParser model = ConsentParser.parse("BObdrPUOevsguAfDqFENCNAAAAAmeAAA");
+        assertTrue(model instanceof ConsentParserV1);
+        assertEquals(1, model.getVersion());
+    }
+
+    @Test
+    public void testCanConstructV1Properties() {
+        ConsentParser model = parse("BObdrPUOevsguAfDqFENCNAAAAAmeAAA");
+        assertEquals(1, model.getVersion());
+        assertEquals(31, model.getCmpId());
+        assertEquals(234, model.getCmpVersion());
+        assertEquals(5, model.getConsentScreen());
+        assertEquals("EN", model.getConsentLanguage());
+        assertEquals(Instant.parse("2019-02-04T21:16:05.200Z"), model.getCreated());
+        assertEquals(Instant.parse("2019-04-09T14:35:10.200Z"), model.getLastUpdated());
+        assertEquals(141, model.getVendorListVersion());
+        assertTrue(model.getPurposesAllowed().isEmpty());
+        assertFalse(model.getPurposesAllowed().contains(1));
+        assertFalse(model.getVendorConsent().contains(1));
+    }
+
+    @Test
+    public void testAllowedPurposes() {
+        ConsentParser model = parse("BOOzQoAOOzQoAAPAFSENCW-AIBA=");
+        assertThat(model.getPurposesAllowed(), matchInts(1, 2, 3, 4, 5, 15, 24));
+        assertTrue(model.getPurposesAllowed().contains(1));
+        assertTrue(model.getPurposesAllowed().contains(STORAGE_AND_ACCESS.getId()));
+        assertTrue(model.getPurposesAllowed().contains(2));
+        assertTrue(model.getPurposesAllowed().contains(PERSONALIZATION.getId()));
+        assertTrue(model.getPurposesAllowed().contains(3));
+        assertTrue(model.getPurposesAllowed().contains(AD_SELECTION.getId()));
+        assertTrue(model.getPurposesAllowed().contains(4));
+        assertTrue(model.getPurposesAllowed().contains(CONTENT_DELIVERY.getId()));
+        assertTrue(model.getPurposesAllowed().contains(5));
+        assertTrue(model.getPurposesAllowed().contains(MEASUREMENT.getId()));
+        assertTrue(model.getPurposesAllowed().contains(15));
+        assertTrue(model.getPurposesAllowed().contains(24));
+    }
+
+    @Test
+    public void testXor() {
+        assertFalse(true ^ true);
+        assertTrue(true ^ false);
+        assertTrue(false ^ true);
+        assertFalse(false ^ false);
+    }
```

Comment:
why we testing that Java's logic is working?

Reply:
this can be removed